src/magic_folder/config.py

```diff
@@ -204,40 +225,102 @@ class MagicFolderConfig(object):
-            cursor.execute("SELECT version FROM version");
-            dbversion = cursor.fetchone()[0]
-            if dbversion != _magicfolder_config_version:
-                raise ConfigurationError(
-                    "Magic Folder '{}' has unknown configuration database "
-                    "version (wanted {}, got {})".format(
-                        self.name,
-                        _magicfolder_config_version,
-                        dbversion,
-                    )
+    @with_cursor
+    def __attrs_post_init__(self, cursor):
+        cursor.execute("BEGIN IMMEDIATE TRANSACTION")
+        cursor.execute("SELECT version FROM version");
+        dbversion = cursor.fetchone()[0]
+        if dbversion != _magicfolder_config_version:
+            raise ConfigurationError(
+                "Magic Folder '{}' has unknown configuration database "
+                "version (wanted {}, got {})".format(
+                    self.name,
+                    _magicfolder_config_version,
+                    dbversion,
                 )
+            )
 
     @property
-    def author(self):
-        with self.database:
-            cursor = self.database.cursor()
-            cursor.execute("SELECT author_name, author_private_key FROM config");
-            name, keydata = cursor.fetchone()
-            return LocalAuthor(
-                name=name,
-                signing_key=SigningKey(keydata, encoder=Base32Encoder),
-            )
+    @with_cursor
+    def author(self, cursor):
+        cursor.execute("SELECT author_name, author_private_key FROM config");
+        name, keydata = cursor.fetchone()
+        return LocalAuthor(
+            name=name,
+            signing_key=SigningKey(keydata, encoder=Base32Encoder),
+        )
 
     @property
-    def stash_path(self):
-        with self.database:
-            cursor = self.database.cursor()
-            cursor.execute("SELECT stash_path FROM config");
-            path_raw = cursor.fetchone()[0]
-            return FilePath(path_raw)
+    @with_cursor
+    def stash_path(self, cursor):
+        cursor.execute("SELECT stash_path FROM config");
+        path_raw = cursor.fetchone()[0]
+        return FilePath(path_raw)
+
+    @with_cursor
+    def get_local_snapshot(self, cursor, name, author):
+        """
+        return an instance of LocalSnapshot corresponding to
+        the given name and author. Traversing the parents
+        would give the entire history of local snapshots.
+
+        :param unicode name: magicpath that represents the relative path of the file.
+
+        :param author: an instance of LocalAuthor
+
+        :returns: An instance of LocalSnapshot for the given magicpath.
+        """
+        cursor.execute("SELECT snapshot_blob FROM local_snapshots"
+                       " WHERE path=?",
+                       (name,))
+        row = cursor.fetchone()
+        if not row:
+            return None
```

Comment:
I guess more of a style/philosophy thing but maybe an exception here is "better"? (e.g. `KeyError` or ..?)

Reply:
Yea, I think an exception is better.  It's harder to accidentally ignore.